### PR TITLE
[Completion] Tweak assert for subscript call pattern

### PR DESCRIPTION
**Pull request by hamishknight**
We may not have a semantic context kind if the subscript decl is missing, for cases such as the implicit `keyPath` subscript.

rdar://139333904

### CODE REVIEW

**Comment by hamishknight** [comment]
@swift-ci please test

**Review by ahoppen — APPROVED** [review]
Nice. That was easy.

**Review by rintaro — APPROVED** [review]

File: test/IDE/complete_subscript.swift [hunk 1]
```diff
@@ -1,23 +1,4 @@
...
-// RUN: %target-swift-ide-test -code-completion -source-filename %s -code-completion-token=SELF_IN_INSTANCEMETHOD | %FileCheck %s -check-prefix=SELF_IN_INSTANCEMETHOD
-// RUN: %target-swift-ide-test -code-completion -source-filename %s -code-completion-token=SUPER_IN_INSTANCEMETHOD | %FileCheck %s -check-prefix=SUPER_IN_INSTANCEMETHOD
-// RUN: %target-swift-ide-test -code-completion -source-filename %s -code-completion-token=SELF_IN_STATICMETHOD | %FileCheck %s -check-prefix=SELF_IN_STATICMETHOD
-// RUN: %target-swift-ide-test -code-completion -source-filename %s -code-completion-token=SUPER_IN_STATICMETHOD | %FileCheck %s -check-prefix=SUPER_IN_STATICMETHOD
-
-// RUN: %target-swift-ide-test -code-completion -source-filename %s -code-completion-token=LABELED_SUBSCRIPT | %FileCheck %s -check-prefix=LABELED_SUBSCRIPT
-// RUN: %target-swift-ide-test -code-completion -source-filename %s -code-completion-token=TUPLE | %FileCheck %s -check-prefix=TUPLE
-// RUN: %target-swift-ide-test -code-completion -source-filename %s -code-completion-token=SETTABLE_SUBSCRIPT | %FileCheck %s -check-prefix=SETTABLE_SUBSCRIPT
+// RUN: %batch-code-completion
```
Comment (bnbarham) [inline test/IDE/complete_subscript.swift:1]: Thank you 🙏

**Review by bnbarham — APPROVED** [review]

**Comment by hamishknight** [comment]
@swift-ci please test macOS

**Pull request merged**